src/js/lib/monster.js

```diff
@@ -474,42 +474,37 @@ define(function(require) {
 			next && next();
 		},
 
-		loadBuildConfig: function(globalCallback) {
+		loadBuildConfig: function(callback) {
 			var self = this;
 
 			monster.parallel({
-				version: function(callback) {
+				version: function(next) {
 					$.ajax({
 						url: 'VERSION',
 						cache: false,
-						success: function(version) {
-							version = version.replace(/\n.*/g, '').trim();
-
-							callback(null, version);
-						},
-						error: function() {
-							callback(null, null);
-						}
+						success: _.partial(next, null),
+						error: _.partial(next, null, null)
 					});
 				},
-				buildFile: function(callback) {
+				buildFile: function(next) {
 					$.ajax({
 						url: 'build-config.json',
 						dataType: 'json',
 						cache: false,
-						success: function(config) {
-							callback(null, config);
-						},
-						error: function() {
-							callback(null, {});
-						}
+						success: _.partial(next, null),
+						error: _.partial(next, null, {})
 					});
 				}
 			}, function(err, results) {
 				monster.config.developerFlags.build = results.buildFile;
-				monster.config.developerFlags.build.version = results.version;
+				monster.config.developerFlags.build.version = _
+					.chain(results.version)
+					.thru(monster.parseVersionFile)
+					.get('version')
+					.defaultTo(null)
+					.value();
 
-				globalCallback && globalCallback(monster.config.developerFlags.build);
+				callback && callback(monster.config.developerFlags.build);
 			});
 		},
```

Comment:
Upon further testing, it seems that the build config is not being loaded properly. Peeking at the `results` variable, I get this object:

![image](https://user-images.githubusercontent.com/893877/106637899-e04b1600-6548-11eb-9f63-7c0c44f4f4b5.png)

```json
{
  "version": [
    "4.3.0",
    "success",
    {
      "readyState": 4,
      "responseText": "4.3.0",
      "status": 200,
      "statusText": "OK"
    }
  ],
  "buildFile": [
    {
      "type": "production",
      "preloadApps": [
        "accounts",
        "appexchange-submission",
        "apploader",
        "appstore",
        "auth",
        "blacklists",
        "branding",
        "call-recording",
        "callerid-manipulation",
        "callflows",
        "callqueues",
        "callqueues-pro",
        "callthru-us",
        "cluster",
        "common",
        "conferences",
        "core",
        "csv-onboarding",
        "desktop-app",
        "dynamic-callerid",
        "mobile",
        "myaccount",
        "numbers",
        "port",
        "service-plan-override",
        "service-planner",
        "userportal",
        "voicemails",
        "voip",
        "webhooks"
      ]
    },
    "success",
    {
      "readyState": 4,
      "responseText": "{\"type\":\"production\",\"preloadApps\":[\"accounts\",\"appexchange-submission\",\"apploader\",\"appstore\",\"auth\",\"blacklists\",\"branding\",\"call-recording\",\"callerid-manipulation\",\"callflows\",\"callqueues\",\"callqueues-pro\",\"callthru-us\",\"cluster\",\"common\",\"conferences\",\"core\",\"csv-onboarding\",\"desktop-app\",\"dynamic-callerid\",\"mobile\",\"myaccount\",\"numbers\",\"port\",\"service-plan-override\",\"service-planner\",\"userportal\",\"voicemails\",\"voip\",\"webhooks\"]}",
      "status": 200,
      "statusText": "OK"
    }
  ]
}
```

Reply:
Hi @guillegr123 are you testing in a local env? I ran my tests in an already deployed for prod env and the VERSION file was updated properly. Will try to run this again.

Reply:
Hi @gerardozaratem Yes, I was testing in a local env, with `gulp serve-prod`

Reply:
Interesting, I've extracted parsing steps into their own function to apply them during the success handler.

Seems to do the the trick.